mlflow/telemetry/parser.py

```diff
@@ -67,28 +63,39 @@ def extract_params(
         return LogModelParams(**record_params)
 
     @classmethod
-    def parse_model_type(cls, model: Any) -> ModelType:
-        try:
-            from mlflow.pyfunc.model import ChatAgent, ChatModel, PythonModel
-        except ImportError:
-            pass
+    def parse_model_type(cls, flavor: str, arguments: dict[str, Any]) -> ModelType:
+        if flavor == "pyfunc":
+            model = arguments.get("python_model")
+            # either python_model or loader_module must be specified
+            if model is None and arguments.get("loader_module"):
+                return ModelType.LOADER_MODULE
+
+            try:
+                from mlflow.pyfunc.model import ResponsesAgent
+            except ImportError:
+                pass
+            else:
+                if isinstance(model, ResponsesAgent):
+                    return ModelType.RESPONSES_AGENT
```

Comment:
nit: can't we include the else block inside try block since this code is executed when `ResponsesAgent` can be imported anyway?